Failure to resolve encrypted configuration properties with the Gradle plugin

### Describe the bug

As of Quarkus `3.5.0` (and also `3.6.4` as I have the same error there), using the encrypted configuration properties as described in the docs https://quarkus.io/guides/config-reference#secret-keys-expressions causes the build to fail when resolving properties, and the application fails to start locally. The exact code linked in the reproducer section works with `3.4.3`.



### Expected behavior

The build should succeed and correctly resolve configuration properties, given that the appropriate configuration dependencies are available on the class path. 

### Actual behavior

When running `./gradlew test`, the build fails with the following error:
```
./gradlew test
> Task :api:quarkusGenerateCode FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':api:quarkusGenerateCode'.
> Error while evaluating property 'cachingRelevantInput' of task ':api:quarkusGenerateCode'.
   > SRCFG00046: Could not find a secret key handler for aes-gcm-nopadding
```
Even though the `smallrye-config-crypto` dependency is on the class path and the `aes-gcm-nopadding` key is configured.


<details>
<summary>The full stack trace can be seen here:</summary>

```
> Task :api:quarkusGenerateCode FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':api:quarkusGenerateCode'.
> Error while evaluating property 'cachingRelevantInput' of task ':api:quarkusGenerateCode'.
   > SRCFG00046: Could not find a secret key handler for aes-gcm-nopadding

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':api:quarkusGenerateCode'.
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
        at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
Caused by: org.gradle.api.internal.tasks.properties.PropertyEvaluationException: Error while evaluating property 'cachingRelevantInput' of task ':api:quarkusGenerateCode'.
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:32)
        at org.gradle.api.internal.tasks.execution.TaskExecution.lambda$visitRegularInputs$1(TaskExecution.java:321)
        at org.gradle.internal.execution.impl.DefaultInputFingerprinter$InputCollectingVisitor.visitInputProperty(DefaultInputFingerprinter.java:103)
        at org.gradle.api.internal.tasks.execution.TaskExecution.visitRegularInputs(TaskExecution.java:319)
        at org.gradle.internal.execution.impl.DefaultInputFingerprinter.fingerprintInputProperties(DefaultInputFingerprinter.java:63)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionStateWithOutputs(CaptureStateBeforeExecutionStep.java:123)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$captureExecutionState$1(CaptureStateBeforeExecutionStep.java:82)
        at org.gradle.internal.execution.steps.BuildOperationStep$1.call(BuildOperationStep.java:37)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.internal.execution.steps.BuildOperationStep.operation(BuildOperationStep.java:34)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionState(CaptureStateBeforeExecutionStep.java:76)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$execute$0(CaptureStateBeforeExecutionStep.java:70)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:70)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:45)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.executeWithNonEmptySources(SkipEmptyWorkStep.java:177)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:81)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:53)
        at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:32)
        at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:21)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
        at org.gradle.internal.execution.steps.CleanupStaleOutputsStep.execute(CleanupStaleOutputsStep.java:75)
        at org.gradle.internal.execution.steps.CleanupStaleOutputsStep.execute(CleanupStaleOutputsStep.java:41)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:66)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:66)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:38)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:32)
        at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:293)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:21)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:47)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:34)
        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:64)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:145)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:134)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
        at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
Caused by: java.util.NoSuchElementException: SRCFG00046: Could not find a secret key handler for aes-gcm-nopadding
        at io.smallrye.config.SecretKeysHandlerConfigSourceInterceptor.getHandler(SecretKeysHandlerConfigSourceInterceptor.java:38)
        at io.smallrye.config.SecretKeysHandlerConfigSourceInterceptor.getValue(SecretKeysHandlerConfigSourceInterceptor.java:29)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
        at io.smallrye.config.FallbackConfigSourceInterceptor.getValue(FallbackConfigSourceInterceptor.java:24)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
        at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:322)
        at io.quarkus.gradle.tasks.EffectiveConfig.lambda$generateFullConfigMap$1(EffectiveConfig.java:110)
        at io.quarkus.gradle.tasks.EffectiveConfig.generateFullConfigMap(EffectiveConfig.java:109)
        at io.quarkus.gradle.tasks.EffectiveConfig.<init>(EffectiveConfig.java:89)
        at io.quarkus.gradle.tasks.EffectiveConfig$Builder.build(EffectiveConfig.java:176)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.buildEffectiveConfiguration(AbstractQuarkusExtension.java:130)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.buildBaseConfig(AbstractQuarkusExtension.java:71)
        at org.gradle.api.internal.provider.DefaultProvider.calculateOwnValue(DefaultProvider.java:72)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:115)
        at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:82)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:115)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.withFinalValue(AbstractMinimalProvider.java:168)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:133)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:26)
        at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:245)
        at org.gradle.api.internal.provider.AbstractProperty.finalizeValue(AbstractProperty.java:194)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.baseConfig(AbstractQuarkusExtension.java:76)
        at io.quarkus.gradle.tasks.QuarkusGenerateCode.getCachingRelevantInput(QuarkusGenerateCode.java:70)
        at io.quarkus.gradle.tasks.QuarkusGenerateCode_Decorated.getCachingRelevantInput(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at org.gradle.internal.reflect.annotations.impl.DefaultPropertyAnnotationMetadata.getPropertyValue(DefaultPropertyAnnotationMetadata.java:97)
        at org.gradle.internal.properties.annotations.DefaultTypeMetadataStore$DefaultPropertyMetadata.getPropertyValue(DefaultTypeMetadataStore.java:288)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$1.lambda$visitLeaf$0(DefaultPropertyWalker.java:90)
        at org.gradle.internal.deprecation.DeprecationLogger.whileDisabled(DeprecationLogger.java:259)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$CachedPropertyValue.lambda$new$0(DefaultPropertyWalker.java:107)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:181)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$CachedPropertyValue.call(DefaultPropertyWalker.java:149)
        at org.gradle.util.internal.GUtil.uncheckedCall(GUtil.java:437)
        at org.gradle.util.internal.DeferredUtil.unpackNestableDeferred(DeferredUtil.java:83)
        at org.gradle.util.internal.DeferredUtil.unpack(DeferredUtil.java:57)
        at org.gradle.util.internal.DeferredUtil.unpackOrNull(DeferredUtil.java:49)
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:38)
        at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:30)
        ... 72 more
```
</details>

### How to Reproduce?

Clone this project: https://github.com/michaeldimchuk/gradle-java-template

It's currently using Quarkus version 3.4.3, which is the latest version of Quarkus which still allows running tests / running the application locally without it crashing. The `api/src/main/resources/application.yaml` file contains the encrypted value that needs to be resolved, which will be loaded as the `StoreSettings` class in the `services` module. With this version of Quarkus, either running `./gradlew test` or `./gradlew quarkusDev` works perfectly fine.

To replicate this being broken, upgrade the following 3 locations to whatever version of Quarkus you want to test with, `3.5.0` is the earliest where this started happening, but I can also replicate this with the latest  `3.6.4`:
- https://github.com/michaeldimchuk/gradle-java-template/blob/main/settings.gradle.kts#L12
- https://github.com/michaeldimchuk/gradle-java-template/blob/main/api/build.gradle.kts#L9
- https://github.com/michaeldimchuk/gradle-java-template/blob/main/services/build.gradle.kts#L9

Then run `make lock` to update the lock files and ensure all dependencies are synchronized.

After this, running either `./gradlew test` or `./gradlew quarkusDev` will fail.


### Output of `uname -a` or `ver`

Darwin Michaels-MacBook-Air.local 22.2.0 Darwin Kernel Version 22.2.0: Fri Nov 11 02:04:44 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T8103 arm64

### Output of `java -version`

openjdk version "21.0.1" 2023-10-17 OpenJDK Runtime Environment Homebrew (build 21.0.1) OpenJDK 64-Bit Server VM Homebrew (build 21.0.1, mixed mode, sharing)

### Quarkus version or git rev

3.5.0 || 3.6.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 8.5 ------------------------------------------------------------  Build time:   2023-11-29 14:08:57 UTC Revision:     28aca86a7180baa17117e0e5ba01d8ea9feca598  Kotlin:       1.9.20 Groovy:       3.0.17 Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023 JVM:          21.0.1 (Homebrew 21.0.1) OS:           Mac OS X 13.1 aarch64

### Additional information

_No response_

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/extension/QuarkusPluginExtension.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/extension/QuarkusPluginExtension.java
@@ -73,7 +73,7 @@ public abstract class QuarkusPluginExtension extends AbstractQuarkusExtension {
             Map<String, Object> props = task.getSystemProperties();
             ApplicationModel appModel = getApplicationModel(TEST);
 
-            SmallRyeConfig config = buildEffectiveConfiguration(appModel.getAppArtifact()).config();
+            SmallRyeConfig config = buildEffectiveConfiguration(appModel.getAppArtifact()).getConfig();
             config.getOptionalValue(TEST.getProfileKey(), String.class)
                     .ifPresent(value -> props.put(TEST.getProfileKey(), value));
 

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/AbstractQuarkusExtension.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/AbstractQuarkusExtension.java
@@ -1,12 +1,17 @@
 package io.quarkus.gradle.tasks;
 
 import static io.quarkus.gradle.tasks.QuarkusGradleUtils.getSourceSet;
+import static io.smallrye.common.expression.Expression.Flag.DOUBLE_COLON;
+import static io.smallrye.common.expression.Expression.Flag.LENIENT_SYNTAX;
+import static io.smallrye.common.expression.Expression.Flag.NO_SMART_BRACES;
+import static io.smallrye.common.expression.Expression.Flag.NO_TRIM;
 import static java.util.Collections.emptyList;
 
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
@@ -24,6 +29,7 @@ import org.gradle.process.JavaForkOptions;
 
 import io.quarkus.gradle.dsl.Manifest;
 import io.quarkus.maven.dependency.ResolvedDependency;
+import io.smallrye.common.expression.Expression;
 
 /**
  * This base class exists to hide internal properties, make those only available in the {@link io.quarkus.gradle.tasks}
@@ -138,7 +144,7 @@ public abstract class AbstractQuarkusExtension {
      * @param appArtifact the application dependency to retrive the quarkus application name and version.
      * @return a filtered view of the configuration only with <code>quarkus.</code> names.
      */
-    protected Map<String, String> buildSystemProperties(ResolvedDependency appArtifact) {
+    protected Map<String, String> buildSystemProperties(ResolvedDependency appArtifact, Map<String, String> quarkusProperties) {
         Map<String, String> buildSystemProperties = new HashMap<>();
         buildSystemProperties.putIfAbsent("quarkus.application.name", appArtifact.getArtifactId());
         buildSystemProperties.putIfAbsent("quarkus.application.version", appArtifact.getVersion());
@@ -158,6 +164,33 @@ public abstract class AbstractQuarkusExtension {
                 buildSystemProperties.put(entry.getKey(), entry.getValue().toString());
             }
         }
+
+        Set<String> quarkusValues = new HashSet<>();
+        quarkusValues.addAll(quarkusProperties.values());
+        quarkusValues.addAll(buildSystemProperties.values());
+
+        for (String value : quarkusValues) {
+            Expression expression = Expression.compile(value, LENIENT_SYNTAX, NO_TRIM, NO_SMART_BRACES, DOUBLE_COLON);
+            for (String reference : expression.getReferencedStrings()) {
+                String expanded = forcedPropertiesProperty.get().get(reference);
+                if (expanded != null) {
+                    buildSystemProperties.put(reference, expanded);
+                    continue;
+                }
+
+                expanded = quarkusBuildProperties.get().get(reference);
+                if (expanded != null) {
+                    buildSystemProperties.put(reference, expanded);
+                    continue;
+                }
+
+                expanded = (String) project.getProperties().get(reference);
+                if (expanded != null) {
+                    buildSystemProperties.put(reference, expanded);
+                }
+            }
+        }
+
         return buildSystemProperties;
     }
 

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/BaseConfig.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/BaseConfig.java
@@ -21,20 +21,20 @@ import io.quarkus.runtime.configuration.ConfigInstantiator;
 final class BaseConfig {
     private final Manifest manifest;
     private final PackageConfig packageConfig;
-    private final Map<String, String> configMap;
+    private final Map<String, String> values;
 
     // Note: EffectiveConfig has all the code to load the configurations from all the sources.
     BaseConfig(EffectiveConfig config) {
         manifest = new Manifest();
         packageConfig = new PackageConfig();
 
-        ConfigInstantiator.handleObject(packageConfig, config.config());
+        ConfigInstantiator.handleObject(packageConfig, config.getConfig());
 
         // populate the Gradle Manifest object
         manifest.attributes(packageConfig.manifest.attributes);
         packageConfig.manifest.manifestSections.forEach((section, attribs) -> manifest.attributes(attribs, section));
 
-        configMap = config.configMap();
+        values = config.getValues();
     }
 
     PackageConfig packageConfig() {
@@ -53,7 +53,7 @@ final class BaseConfig {
         List<Pattern> patterns = propertyPatterns.stream().map(s -> "^(" + s + ")$").map(Pattern::compile)
                 .collect(Collectors.toList());
         Predicate<Map.Entry<String, ?>> keyPredicate = e -> patterns.stream().anyMatch(p -> p.matcher(e.getKey()).matches());
-        return configMap.entrySet().stream()
+        return values.entrySet().stream()
                 .filter(keyPredicate)
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/Deploy.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/Deploy.java
@@ -90,9 +90,8 @@ public abstract class Deploy extends QuarkusBuildTask {
     @TaskAction
     public void checkRequiredExtensions() {
         ApplicationModel appModel = resolveAppModelForBuild();
-        Map<String, String> configMap = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).configMap();
         Properties sysProps = new Properties();
-        sysProps.putAll(configMap);
+        sysProps.putAll(extension().buildEffectiveConfiguration(appModel.getAppArtifact()).getValues());
         try (CuratedApplication curatedApplication = QuarkusBootstrap.builder()
                 .setBaseClassLoader(getClass().getClassLoader())
                 .setExistingModel(appModel)

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/EffectiveConfig.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/EffectiveConfig.java
@@ -14,6 +14,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Supplier;
 
 import org.eclipse.microprofile.config.spi.ConfigSource;
 import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
@@ -26,6 +27,7 @@ import io.quarkus.runtime.configuration.ApplicationPropertiesConfigSourceLoader;
 import io.quarkus.runtime.configuration.ConfigUtils;
 import io.smallrye.config.AbstractLocationConfigSourceLoader;
 import io.smallrye.config.EnvConfigSource;
+import io.smallrye.config.Expressions;
 import io.smallrye.config.PropertiesConfigSource;
 import io.smallrye.config.PropertiesConfigSourceProvider;
 import io.smallrye.config.SmallRyeConfig;
@@ -42,9 +44,8 @@ import io.smallrye.config.source.yaml.YamlConfigSource;
  * the Quarkus config objects like {@link PackageConfig}, {@link ClassLoadingConfig} and the underlying {@link SmallRyeConfig}.
  */
 public final class EffectiveConfig {
-    private final Map<String, String> fullConfig;
-
     private final SmallRyeConfig config;
+    private final Map<String, String> values;
 
     private EffectiveConfig(Builder builder) {
         List<ConfigSource> configSources = new ArrayList<>();
@@ -81,11 +82,15 @@ public final class EffectiveConfig {
                 .addAll(PropertiesConfigSourceProvider.classPathSources(META_INF_MICROPROFILE_CONFIG_PROPERTIES, classLoader));
 
         this.config = buildConfig(builder.profile, configSources);
-        this.fullConfig = generateFullConfigMap(config);
+        this.values = generateFullConfigMap(config);
     }
 
-    public SmallRyeConfig config() {
+    public SmallRyeConfig getConfig() {
         return config;
+    }
+
+    public Map<String, String> getValues() {
+        return values;
     }
 
     private Map<String, String> asStringMap(Map<String, ?> map) {
@@ -100,14 +105,19 @@ public final class EffectiveConfig {
 
     @VisibleForTesting
     static Map<String, String> generateFullConfigMap(SmallRyeConfig config) {
-        Map<String, String> map = new HashMap<>();
-        config.getPropertyNames().forEach(property -> {
-            String v = config.getConfigValue(property).getValue();
-            if (v != null) {
-                map.put(property, v);
+        return Expressions.withoutExpansion(new Supplier<Map<String, String>>() {
+            @Override
+            public Map<String, String> get() {
+                Map<String, String> properties = new HashMap<>();
+                for (String propertyName : config.getPropertyNames()) {
+                    String value = config.getRawValue(propertyName);
+                    if (value != null) {
+                        properties.put(propertyName, value);
+                    }
+                }
+                return unmodifiableMap(properties);
             }
         });
-        return unmodifiableMap(map);
     }
 
     @VisibleForTesting
@@ -124,10 +134,6 @@ public final class EffectiveConfig {
 
     static Builder builder() {
         return new Builder();
-    }
-
-    public Map<String, String> configMap() {
-        return fullConfig;
     }
 
     static final class Builder {

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildDependencies.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildDependencies.java
@@ -24,6 +24,7 @@ import io.quarkus.gradle.QuarkusPlugin;
 import io.quarkus.maven.dependency.ArtifactKey;
 import io.quarkus.maven.dependency.DependencyFlags;
 import io.quarkus.maven.dependency.ResolvedDependency;
+import io.smallrye.config.SmallRyeConfig;
 
 /**
  * Collect the Quarkus app dependencies, the contents of the {@code quarkus-app/lib} folder, without making the task
@@ -139,22 +140,23 @@ public abstract class QuarkusBuildDependencies extends QuarkusBuildTask {
         }
 
         ApplicationModel appModel = resolveAppModelForBuild();
-        Map<String, String> configMap = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).configMap();
+        SmallRyeConfig config = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).getConfig();
 
         // see https://quarkus.io/guides/class-loading-reference#configuring-class-loading
-        Set<ArtifactKey> removedArtifacts = java.util.Optional.ofNullable(
-                configMap.getOrDefault(CLASS_LOADING_REMOVED_ARTIFACTS, null))
+        Set<ArtifactKey> removedArtifacts = config.getOptionalValue(CLASS_LOADING_REMOVED_ARTIFACTS, String.class)
                 .map(QuarkusBuildDependencies::dependenciesListToArtifactKeySet)
                 .orElse(Collections.emptySet());
-        getLogger().info("Removed artifacts: {}", configMap.getOrDefault(CLASS_LOADING_REMOVED_ARTIFACTS, "(none)"));
+        getLogger().info("Removed artifacts: {}",
+                config.getOptionalValue(CLASS_LOADING_REMOVED_ARTIFACTS, String.class).orElse("(none)"));
 
-        String parentFirstArtifactsProp = configMap.getOrDefault(CLASS_LOADING_PARENT_FIRST_ARTIFACTS, "");
+        String parentFirstArtifactsProp = config.getOptionalValue(CLASS_LOADING_PARENT_FIRST_ARTIFACTS, String.class)
+                .orElse("");
         Set<ArtifactKey> parentFirstArtifacts = dependenciesListToArtifactKeySet(parentFirstArtifactsProp);
-        getLogger().info("parent first artifacts: {}", configMap.getOrDefault(CLASS_LOADING_PARENT_FIRST_ARTIFACTS, "(none)"));
+        getLogger().info("parent first artifacts: {}",
+                config.getOptionalValue(CLASS_LOADING_PARENT_FIRST_ARTIFACTS, String.class).orElse("(none)"));
 
-        String optionalDependenciesProp = configMap.getOrDefault(INCLUDED_OPTIONAL_DEPENDENCIES, "");
-        boolean filterOptionalDependencies = Boolean
-                .parseBoolean(configMap.getOrDefault(FILTER_OPTIONAL_DEPENDENCIES, "false"));
+        String optionalDependenciesProp = config.getOptionalValue(INCLUDED_OPTIONAL_DEPENDENCIES, String.class).orElse("");
+        boolean filterOptionalDependencies = config.getOptionalValue(FILTER_OPTIONAL_DEPENDENCIES, Boolean.class).orElse(false);
         Set<ArtifactKey> optionalDependencies = filterOptionalDependencies
                 ? dependenciesListToArtifactKeySet(optionalDependenciesProp)
                 : Collections.emptySet();

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
@@ -29,6 +29,7 @@ import io.quarkus.gradle.QuarkusPlugin;
 import io.quarkus.gradle.tasks.worker.BuildWorker;
 import io.quarkus.maven.dependency.GACTV;
 import io.smallrye.config.Expressions;
+import io.smallrye.config.SmallRyeConfig;
 
 /**
  * Base class for the {@link QuarkusBuildDependencies}, {@link QuarkusBuildCacheableAppParts}, {@link QuarkusBuild} tasks
@@ -207,30 +208,29 @@ abstract class QuarkusBuildTask extends QuarkusTask {
         });
 
         ApplicationModel appModel = resolveAppModelForBuild();
-        Map<String, String> configMap = new HashMap<>();
-        EffectiveConfig effectiveConfig = extension().buildEffectiveConfiguration(appModel.getAppArtifact());
-        Expressions.withoutExpansion(() -> {
-            for (Map.Entry<String, String> entry : effectiveConfig.configMap().entrySet()) {
-                if (entry.getKey().startsWith("quarkus.")) {
-                    configMap.put(entry.getKey(), effectiveConfig.config().getRawValue(entry.getKey()));
-                }
-            }
+        SmallRyeConfig config = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).getConfig();
+        Map<String, String> quarkusProperties = Expressions.withoutExpansion(() -> {
+            Map<String, String> values = new HashMap<>();
+            config.getValues("quarkus", String.class, String.class)
+                    .forEach((key, value) -> values.put("quarkus." + key, value));
+            return values;
         });
 
         getLogger().info("Starting Quarkus application build for package type {}", packageType);
 
         if (getLogger().isEnabled(LogLevel.INFO)) {
             getLogger().info("Effective properties: {}",
-                    configMap.entrySet().stream()
+                    quarkusProperties.entrySet().stream()
                             .map(Object::toString)
                             .sorted()
                             .collect(Collectors.joining("\n    ", "\n    ", "")));
         }
 
-        WorkQueue workQueue = workQueue(configMap, () -> extension().buildForkOptions);
+        WorkQueue workQueue = workQueue(quarkusProperties, () -> extension().buildForkOptions);
 
         workQueue.submit(BuildWorker.class, params -> {
-            params.getBuildSystemProperties().putAll(extension().buildSystemProperties(appModel.getAppArtifact()));
+            params.getBuildSystemProperties()
+                    .putAll(extension().buildSystemProperties(appModel.getAppArtifact(), quarkusProperties));
             params.getBaseName().set(extension().finalName());
             params.getTargetDirectory().set(buildDir.toFile());
             params.getAppModel().set(appModel);

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java
@@ -102,17 +102,17 @@ public abstract class QuarkusGenerateCode extends QuarkusTask {
     @TaskAction
     public void generateCode() {
         ApplicationModel appModel = extension().getApplicationModel(launchMode);
-        Map<String, String> configMap = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).configMap();
+        Map<String, String> values = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).getValues();
 
         File outputPath = getGeneratedOutputDirectory().get().getAsFile();
 
         getLogger().debug("Will trigger preparing sources for source directories: {} buildDir: {}",
                 sourcesDirectories, buildDir.getAbsolutePath());
 
-        WorkQueue workQueue = workQueue(configMap, () -> extension().codeGenForkOptions);
+        WorkQueue workQueue = workQueue(values, () -> extension().codeGenForkOptions);
 
         workQueue.submit(CodeGenWorker.class, params -> {
-            params.getBuildSystemProperties().putAll(configMap);
+            params.getBuildSystemProperties().putAll(values);
             params.getBaseName().set(extension().finalName());
             params.getTargetDirectory().set(buildDir);
             params.getAppModel().set(appModel);

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusRun.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusRun.java
@@ -100,9 +100,8 @@ public abstract class QuarkusRun extends QuarkusBuildTask {
     @TaskAction
     public void runQuarkus() {
         ApplicationModel appModel = resolveAppModelForBuild();
-        Map<String, String> configMap = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).configMap();
         Properties sysProps = new Properties();
-        sysProps.putAll(configMap);
+        sysProps.putAll(extension().buildEffectiveConfiguration(appModel.getAppArtifact()).getValues());
         try (CuratedApplication curatedApplication = QuarkusBootstrap.builder()
                 .setBaseClassLoader(getClass().getClassLoader())
                 .setExistingModel(appModel)

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusShowEffectiveConfig.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusShowEffectiveConfig.java
@@ -9,7 +9,6 @@ import java.io.IOException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import java.util.Properties;
 import java.util.stream.Collectors;
 
@@ -22,6 +21,7 @@ import org.gradle.api.tasks.TaskAction;
 import org.gradle.api.tasks.options.Option;
 
 import io.quarkus.gradle.QuarkusPlugin;
+import io.smallrye.config.SmallRyeConfig;
 
 /**
  * Just show the effective configuration and settings.
@@ -45,21 +45,21 @@ public abstract class QuarkusShowEffectiveConfig extends QuarkusBuildTask {
     @TaskAction
     public void dumpEffectiveConfiguration() {
         try {
-            EffectiveConfig effective = extension()
+            EffectiveConfig effectiveConfig = extension()
                     .buildEffectiveConfiguration(extension().getApplicationModel().getAppArtifact());
-            Map<String, String> configMap = effective.configMap();
+            SmallRyeConfig config = effectiveConfig.getConfig();
             List<String> sourceNames = new ArrayList<>();
-            effective.config().getConfigSources().forEach(configSource -> sourceNames.add(configSource.getName()));
+            config.getConfigSources().forEach(configSource -> sourceNames.add(configSource.getName()));
 
-            String config = configMap.entrySet().stream()
-                    .filter(e -> e.getKey().startsWith("quarkus."))
-                    .map(e -> format("%s=%s", e.getKey(), e.getValue())).sorted()
+            String quarkusConfig = config.getValues("quarkus", String.class, String.class)
+                    .entrySet()
+                    .stream()
+                    .map(e -> format("quarkus.%s=%s", e.getKey(), e.getValue())).sorted()
                     .collect(Collectors.joining("\n    ", "\n    ", "\n"));
-
-            getLogger().lifecycle("Effective Quarkus configuration options: {}", config);
+            getLogger().lifecycle("Effective Quarkus configuration options: {}", quarkusConfig);
 
             String finalName = extension().finalName();
-            String packageType = configMap.getOrDefault(QuarkusPlugin.QUARKUS_PACKAGE_TYPE, "fast-jar");
+            String packageType = config.getOptionalValue(QuarkusPlugin.QUARKUS_PACKAGE_TYPE, String.class).orElse("fast-jar");
             File fastJar = fastJar();
             getLogger().lifecycle(
                     "Quarkus package type:          {}\n" +
@@ -79,7 +79,7 @@ public abstract class QuarkusShowEffectiveConfig extends QuarkusBuildTask {
 
             if (getSaveConfigProperties().get()) {
                 Properties props = new Properties();
-                props.putAll(configMap);
+                props.putAll(effectiveConfig.getValues());
                 Path file = buildDir.toPath().resolve(finalName + ".quarkus-build.properties");
                 try (BufferedWriter writer = newBufferedWriter(file)) {
                     props.store(writer, format("Quarkus build properties with package type %s", packageType));

--- a/devtools/gradle/gradle-application-plugin/src/test/java/io/quarkus/gradle/tasks/EffectiveConfigTest.java
+++ b/devtools/gradle/gradle-application-plugin/src/test/java/io/quarkus/gradle/tasks/EffectiveConfigTest.java
@@ -33,21 +33,21 @@ public class EffectiveConfigTest {
 
         // Cannot do an exact match, because `map` contains both the "raw" environment variables AND the
         // "property-key-ish" entries - i.e. environment appears "twice".
-        soft.assertThat(effectiveConfig.configMap()).containsAllEntriesOf(expect);
+        soft.assertThat(effectiveConfig.getValues()).containsAllEntriesOf(expect);
     }
 
     @Test
     void fromProjectProperties() {
         EffectiveConfig effectiveConfig = EffectiveConfig.builder().withProjectProperties(Map.of("quarkus.foo", "bar")).build();
 
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.foo", "bar");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.foo", "bar");
     }
 
     @Test
     void fromForcedProperties() {
         EffectiveConfig effectiveConfig = EffectiveConfig.builder().withTaskProperties(Map.of("quarkus.foo", "bar")).build();
 
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.foo", "bar");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.foo", "bar");
     }
 
     @Test
@@ -56,7 +56,7 @@ public class EffectiveConfigTest {
         EffectiveConfig effectiveConfig = EffectiveConfig.builder()
                 .withTaskProperties(Map.of("quarkus.foo", "${aes-gcm-nopadding::superSecret}")).build();
 
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.foo", "superSecret");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.foo", "superSecret");
     }
 
     @Test
@@ -69,13 +69,13 @@ public class EffectiveConfigTest {
 
         EffectiveConfig effectiveConfig = EffectiveConfig.builder().withSourceDirectories(source).build();
 
-        SmallRyeConfig config = effectiveConfig.config();
+        SmallRyeConfig config = effectiveConfig.getConfig();
         List<String> sourceNames = new ArrayList<>();
         config.getConfigSources().forEach(configSource -> sourceNames.add(configSource.getName()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url1.getPath()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url2.getPath()));
         // The YAML source is always higher in ordinal than the properties source
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.prop.overload", "from-yaml");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.prop.overload", "from-yaml");
     }
 
     @Test
@@ -90,14 +90,14 @@ public class EffectiveConfigTest {
 
         EffectiveConfig effectiveConfig = EffectiveConfig.builder().withSourceDirectories(source).build();
 
-        SmallRyeConfig config = effectiveConfig.config();
+        SmallRyeConfig config = effectiveConfig.getConfig();
         List<String> sourceNames = new ArrayList<>();
         config.getConfigSources().forEach(configSource -> sourceNames.add(configSource.getName()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url1.getPath()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url2.getPath()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url3.getPath()));
         // The YAML source is always higher in ordinal than the properties source
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.prop.overload", "from-yaml");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.prop.overload", "from-yaml");
     }
 
     @Test
@@ -116,7 +116,7 @@ public class EffectiveConfigTest {
 
         EffectiveConfig effectiveConfig = EffectiveConfig.builder().withSourceDirectories(source).build();
 
-        SmallRyeConfig config = effectiveConfig.config();
+        SmallRyeConfig config = effectiveConfig.getConfig();
         List<String> sourceNames = new ArrayList<>();
         config.getConfigSources().forEach(configSource -> sourceNames.add(configSource.getName()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url1.getPath()));
@@ -125,6 +125,6 @@ public class EffectiveConfigTest {
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url4.getPath()));
         soft.assertThat(sourceNames).anyMatch(s -> s.contains(url5.getPath()));
         // The YAML source is always higher in ordinal than the properties source, even for profile property names
-        soft.assertThat(effectiveConfig.configMap()).containsEntry("quarkus.prop.overload", "from-yaml-prod");
+        soft.assertThat(effectiveConfig.getValues()).containsEntry("quarkus.prop.overload", "from-yaml-prod");
     }
 }

